File: app/src/main/java/com/saint/struct/StructApp.java
package com.saint.struct;

import android.app.Application;

import com.saint.struct.tool.CrashHandler;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class StructApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
//        CrashHandler crashHandler = new CrashHandler();
//        Thread.setDefaultUncaughtExceptionHandler(crashHandler);
    }
}

File: app/src/main/java/com/saint/struct/StructApp.kt
package com.saint.struct

import android.app.Application


class StructApp : Application() {

    override fun onCreate() {
        super.onCreate()
        application = this;
//        val crashHandler = CrashHandler();
//        Thread.setDefaultUncaughtExceptionHandler(crashHandler);
//        Looper.getMainLooper().setMessageLogging { s -> Log.v("debug", s) }
    }

    fun getApplication(): StructApp {
        return application
    }

    companion object {
        lateinit var application: StructApp
    }
}